lib/mongoid/persistable/settable.rb

```diff
@@ -25,15 +25,15 @@ def set(setters)
 
             field_and_value_hash = hasherizer(field.split('.'), value)
             field = field_and_value_hash.keys.first.to_s
+            value = field_and_value_hash[field]
 
-            if fields[field] && fields[field].type == Hash && attributes.key?(field) && !value.empty?
+            if fields[field] && fields[field].type == Hash && attributes.key?(field) && Hash === value && !value.empty?
```

Comment:
Do you think `respond_to?(:empty?)` would make more sense here?

Reply:
Not sure. The behavior in this PR seems to make sense: only merge an existing Hash with a new value if that value is a non-empty Hash. The PR that introduced the regression intended to support setting field values to an empty Hash, which remains supported.